compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -6229,10 +6227,7 @@ OMR::Z::TreeEvaluator::bstoreEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    // If the only consumer is the bstore, then don't bother extending
    //
    else if (valueChild->getReferenceCount() == 1 && valueChild->getRegister() == NULL &&
-             (valueChild->getOpCodeValue() == TR::buload  ||
-              valueChild->getOpCodeValue() == TR::bload   ||
-              valueChild->getOpCodeValue() == TR::buloadi ||
-              valueChild->getOpCodeValue() == TR::buload      ))
+             valueChild->getOpCodeValue() == TR::bload   )
       {
       sourceRegister = cg->allocateRegister();
       TR::MemoryReference * tempMR2 = generateS390MemoryReference(valueChild, cg);
```

Comment:
I think this optimization should work for `bloadi` as well. However this is outside the scope of this PR. I've opened up to investigate.

Reply:
not sure it is beyond the scope because the buloadi case would convert to bloadi no?

Reply:
I think it _should_, I was just being cautious as it would be enabling a path which would not have been taken today for `bloadi`.

Reply:
This was a mistake while I tried to map the `buloadi`/`buload` to their signed ILOpcodes. `buloadi` should be converted to `bloadi`. I have fixed this in commit 9e5a25a